### DISCUSSION TITLE
https in adonis v5

### DISCUSSION THREAD

**Discussion by birutaibm** [adonisjs/core#1695] [Help]
Hi, I create a simple API with adonis v5 and want to use https instead of http when running at localhost.
I found an example of this using previous version of adonis [here](https://adonisjs.com/recipes/4.1/using-https#_self_signed_certificate), but nothing with v5.
I try to use the same code, but it seems `Ignitor` has needer `appRoot` nor `fireHttpServer` at this version.
Is there some example or documentation of adonis v5 using https?

**Comment by thetutlage** [comment]
You can pass a callback to the `start` method in v5.

```ts
new Ignitor(__dirname)
  .httpServer()
  .start((handle) => {
    return createServer(httpsOptions, handle)
  })
```

**Reply by Mint-Joy** [reply]
All subject in [https://github.com/adonisjs/core/discussions/1742](https://github.com/adonisjs/core/discussions/1742)
